File: problems/1. Two Sum/main.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        table = {}
        for i, value in enumerate(nums):
            if target - value in table:
                return (i, table[target - value])
            table[value] = i


nums = [2, 7, 11, 15]
target = 9
result = Solution().twoSum(nums, target)

File: problems/1. Two Sum/main.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
  const table = new Map();

  for (let i = 0; i < nums.length; i++) {
    const complement = target - nums[i];

    if (table.has(complement)) {
      return [table.get(complement), i];
    }

    table.set(nums[i], i);
  }
};
